### PR TITLE
IAM EOF to Data Block

### PR DESCRIPTION
**Pull request by wesleykirklandsg**
IAM EOF to Data Block

### CODE REVIEW

File: iamdata.tf [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    ]
+
+    effect = "Allow"
+
+    resources = [
+      "${aws_iam_role.this.arn}"
+    ]
+
+    condition {
+      test     = "BoolIfExists"
```
Comment (rchildress87) [inline iamdata.tf:120]: Think you want `bool` here
https://github.com/StratusGrid/terraform-aws-iam-group-restricted-admin/pull/2

File: iamdata.tf [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    sid = "AssumeRole"
+
+    actions = [
+        "sts:AssumeRole"
+    ]
+
+    effect = "Allow"
+
+    resources = [
+      "${aws_iam_role.this.arn}"
```
Comment (rchildress87) [inline iamdata.tf:116]: Interpolation-only syntax

**Review by rchildress87 — CHANGES_REQUESTED** [review]

File: iamdata.tf [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      values = [
+        "true"
+      ]
+    }
+  }
+
+  statement {
+    sid = "DenyAllExceptListedIfNoMFA"
+
+    actions = [
```
[the remark this answers is not in the document]
Comment (wesleykirklandsg) [reply iamdata.tf]: Resolved in 6ffc288

File: iamdata.tf (hunk 2 above)
Comment (wesleykirklandsg) [reply iamdata.tf]: Corrected under ca22c5f

File: iamdata.tf (hunk 1 above)
Comment (rchildress87) [reply iamdata.tf:120]: I think one of the bool example under Alternative Combinations should be considered as better option but, according to above documentation, this does emulate the policy that is created when you use the IAM visual editor and choose MFA Required. It also matches the policy that this data source is replacing. I'll approve.

**Review by rchildress87 — APPROVED** [review]

**Pull request merged**